tests/jballerina-unit-test/src/test/resources/test-src/object/anon_object.bal

```diff
@@ -292,3 +292,23 @@ function testCodeAnalyzerRunningOnAnonymousObjectsForDeprecatedFunctionAnnotatio
 isolated function Test() returns int {
     return 0;
 }
+
+function testLocalVariablesFromFields() {
+    int a = 10;
+
+    var _ = object {
+        int x = a;
+
+        function init() {
+            assertEquality(self.x, 10);
```

Comment:
This test relies on the init being called. IMO, it would be better to do the assertion outside: field access on the object constructed.

Reply:
Changed